File: cpp/1259.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
    vector<int> even;
    vector<int> odd;
    int n, number;
    cin >> n;

    for (n; n > 0; n--) {
        cin >> number;

        if (number % 2 == 0) {
            even.push_back(number);
        } else {
            odd.push_back(number);
        }
    }

    sort(even.begin(), even.end());
    sort(odd.begin(), odd.end(), greater<int>());

    for (auto i: even) {
        cout << i << '\n';
    }

    for (auto i: odd) {
        cout << i << '\n';
    }

    return 0;
}

File: py/1259.py
n = int(input())

numbers = []
while n > 0:
    numbers.append(int(input()))
    n -= 1

even = [i for i in numbers if i % 2 == 0]
odd = [i for i in numbers if i % 2 != 0]

even.sort()
odd.sort(reverse=True)

print(*even, sep='\n')
print(*odd, sep='\n')